core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/workflow/impl/ProgressImpl.java

```diff
@@ -0,0 +1,156 @@
+  private User user;
+
+  @Temporal(TemporalType.TIMESTAMP)
+  @Column(name = "created", nullable = false, updatable = false)
+  private Date creationTime;
+
+  @Temporal(TemporalType.TIMESTAMP)
+  @Column(name = "lastModified", nullable = false)
+  private Date lastModified;
+
+  @OneToMany(targetEntity = AbstractProgressStep.class, mappedBy = "id.progress")
+  @Sort(type = SortType.COMPARATOR, comparator = ProgressStepComparator.class)
+  private SortedSet<ProgressStep> steps;
+
+  @Override
+  public long getId() {
+    return workflowProgressId;
+  }
+
+  @Override
+  public void setId(long id) {
+    this.workflowProgressId = id;
+  }
+
+  @Override
+  public WorkflowName getWorkflowName() {
+    return workflowName;
+  }
+
+  @Override
+  public void setWorkflowName(WorkflowName workflowName) {
+    this.workflowName = workflowName;
+  }
+
+  @Override
+  public User getUser() {
+    return user;
+  }
+
+  @Override
+  public void setUser(User user) {
+    this.user = user;
+  }
+
+  @Override
+  public Date getCreationTime() {
+    return creationTime;
+  }
+
+  @Override
+  public void setCreationTime(Date creationTime) {
+    this.creationTime = creationTime;
+  }
+
+  @Override
+  public Date getLastModified() {
+    return lastModified;
+  }
+
+  @Override
+  public void setLastModified(Date lastModified) {
+    this.lastModified = lastModified;
+  }
+
+  @Override
+  public List<ProgressStep> getSteps() {
+    return new ArrayList<>(steps);
+  }
+
+  @Override
+  public void setSteps(List<ProgressStep> steps) {
+    this.steps = new TreeSet<>(new ProgressStepComparator());
+    this.steps.addAll(steps);
+  }
+
+  @Override
+  public int hashCode() {
+    final int prime = 31;
+    int result = 1;
+    result = prime * result + ((creationTime == null) ? 0 : creationTime.hashCode());
+    result = prime * result + ((lastModified == null) ? 0 : lastModified.hashCode());
+    result = prime * result + ((user == null) ? 0 : user.hashCode());
+    result = prime * result + (int) (workflowProgressId ^ (workflowProgressId >>> 32));
+    result = prime * result + ((workflowName == null) ? 0 : workflowName.hashCode());
+    return result;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (this == obj) return true;
+    if (obj == null) return false;
+    if (getClass() != obj.getClass()) return false;
+    ProgressImpl other = (ProgressImpl) obj;
+    if (creationTime == null) {
+      if (other.creationTime != null) return false;
+    } else if (!creationTime.equals(other.creationTime)) return false;
+    if (lastModified == null) {
+      if (other.lastModified != null) return false;
+    } else if (!lastModified.equals(other.lastModified)) return false;
+    if (user == null) {
+      if (other.user != null) return false;
+    } else if (!user.equals(other.user)) return false;
+    if (workflowProgressId != other.workflowProgressId) return false;
+    if (workflowName != other.workflowName) return false;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1126)